pyogrio/_io.pyx

```diff
@@ -793,9 +813,9 @@ cdef validate_feature_range(
         return 0, 0
 
     if skip_features >= feature_count:
-        skip_features = feature_count
+        return 0, 0
```

Comment:
This is the main fix: if `skip_features` is larger than the feature count, we want to return an empty result, so just set skip_features and max_features both to 0 as the simplest way to get an empty result.